release_docs/RELEASE.txt

```diff
@@ -1029,10 +1070,67 @@ Bug Fixes since HDF5-1.14.3 release
 
     Tools
     -----
+    - Fixed several issues in ph5diff
+
+      The parallel logic for the ph5diff tool inside the shared h5diff code was
+      refactored and cleaned up to fix several issues with the ph5diff tool. This
+      fixed:
+
+        - several concurrency issues in ph5diff that can result in interleaved
+          output
+        - an issue where output can sometimes be dropped when it ends up in
+          ph5diff's output overflow file
+        - an issue where MPI_Init was called after HDF5 had been initialized,
+          preventing the library from setting up an MPI communicator attribute
+          to perform library cleanup on MPI_Finalize
+
     - Renamed h5fuse.sh to h5fuse
 
       Addresses Discussion #3791
 
+    - Fixed an issue with unmatched MPI messages in ph5diff
```

Comment:
At least this and the next item below are old fixes in previous releases. I can see the entries in the 1.14.2 and 1.14.3 RELEASE.txt files, but the entries don't appear to be in either the RELEASE.txt or HISTORY-1_14.txt files in the hdf5_1_14 branch.

Reply:
I see "Fixed an issue with unmatched MPI messages in ph5diff" in the 1.14.3 section and "Fixed an issue in h5repack for variable-length typed datasets" in the 1.14.2 section of HISTORY-1_14.txt in the hdf5_1_14 branch.  I'll remove them - I probably made an incorrect conflict resolution choice.